### PR TITLE
MM-14273: Including documentation for bool element, also know as checkbox. 

### PR DESCRIPTION
**Pull request by hvhallmann**
This element enables integration using checkbox for true or false selection.

### CODE REVIEW

**Comment by hvhallmann** [comment]
This PR is mentioned on the following project link
https://github.com/mattermost/mattermost-webapp/pull/3066

File: source/developer/interactive-dialogs.rst [hunk 1]
```diff
@@ -210,6 +210,36 @@ The list of supported fields for the ``select`` type element is included below:
...
+.. csv-table::
+    :header: "Field", "Type", "Description"
+
+    "display_name", "String", "Display name of the field shown to the user in the dialog. Maximum 24 characters."
+    "name", "String", "Name of the field element used by the integration. Maximum 300 characters. You should use unique “name” fields in the same dialog."
+    "type", "String", "Set this value to ``bool`` for a checkbox element."
+    "optional", "Boolean", "(Optional) Set to true if this form element is not required. Default is ``false``."
+    "help_text", "String", "(Optional) Set help text for this form element. Maximum 150 characters."
+    "default", "String", "(Optional) Set a default value for this form element. True or false"
+    "placeholder", "String", "(Optional) A string displayed to include a label besides the checkbox. Maximum 150 characters."
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst]: What is the difference between `placeholder` and `help_text`?

File: source/developer/interactive-dialogs.rst [hunk 2]
```diff
@@ -210,6 +210,36 @@ The list of supported fields for the ``select`` type element is included below:
     "default", "String", "(Optional) Set a default value for this form element. Maximum 3,000 characters."
     "placeholder", "String", "(Optional) A string displayed to help guide users in completing the element. Maximum 3,000 characters."
 
+Checkbox Element
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Checkbox element are single-line plain text fields with a Checkbox to be selected. Below is an example of a Checkbox element that asks for meeting feedback.
```
Comment (hanzei) [inline source/developer/interactive-dialogs.rst]: Maybe the required Mattermost Version (5.12) should be documented?

File: source/developer/interactive-dialogs.rst (hunk 1 above)
Comment (ali-farooq0) [reply source/developer/interactive-dialogs.rst]: I believe `help_test` appears below the field while `placeholder` appears to the right of the checkbox.

as per https://github.com/mattermost/mattermost-webapp/blob/master/components/widgets/settings/setting.jsx#L29

and https://github.com/mattermost/mattermost-webapp/pull/3066/files#diff-c3112a2a9bea3269d2250feb71051a89R47

Comment (hanzei) [reply source/developer/interactive-dialogs.rst]: I leaved a comment about this in the original PR:
https://github.com/mattermost/mattermost-webapp/pull/3066#discussion_r303875873

File: source/developer/interactive-dialogs.rst [hunk 3]
```diff
@@ -210,6 +210,36 @@ The list of supported fields for the ``select`` type element is included below:
...
+
+.. csv-table::
+    :header: "Field", "Type", "Description"
+
+    "display_name", "String", "Display name of the field shown to the user in the dialog. Maximum 24 characters."
+    "name", "String", "Name of the field element used by the integration. Maximum 300 characters. You should use unique “name” fields in the same dialog."
+    "type", "String", "Set this value to ``bool`` for a checkbox element."
+    "optional", "Boolean", "(Optional) Set to true if this form element is not required. Default is ``false``."
+    "help_text", "String", "(Optional) Set help text for this form element. Maximum 150 characters."
+    "default", "String", "(Optional) Set a default value for this form element. True or false"
```
Comment (amyblais) [inline source/developer/interactive-dialogs.rst]:
```suggestion
    "default", "String", "(Optional) Set a default value for this form element. True or false".
```

File: source/developer/interactive-dialogs.rst [hunk 4]
```diff
@@ -210,6 +210,36 @@ The list of supported fields for the ``select`` type element is included below:
     "default", "String", "(Optional) Set a default value for this form element. Maximum 3,000 characters."
     "placeholder", "String", "(Optional) A string displayed to help guide users in completing the element. Maximum 3,000 characters."
 
+Checkbox Element
+^^^^^^^^^^^^^^^^^^^^^^^
+
+In Mattermost version 5.14 and later, you can use Checkbox elements. It looks like a plain text field with a Checkbox to be selected. Below is an example of a Checkbox element that asks for meeting feedback.
```
Comment (amyblais) [inline source/developer/interactive-dialogs.rst]:
```suggestion
In Mattermost version 5.14 and later, you can use Checkbox elements. It looks like a plain text field with a Checkbox to be selected. Below is an example of a Checkbox element that asks for meeting feedback:
```

Comment (hanzei) [inline source/developer/interactive-dialogs.rst]: This needs to be `5.16`

File: source/developer/interactive-dialogs.rst [hunk 5]
```diff
@@ -210,6 +210,36 @@ The list of supported fields for the ``select`` type element is included below:
...
+  {
+    "display_name": "Can you please select below",
+    "placeholder": "The meeting was helpful.",
+    "name": "meeting_input",
+    "type": "bool",
+  }
+
+The full list of supported fields are included below:
+
+.. csv-table::
```
Comment (amyblais) [inline source/developer/interactive-dialogs.rst]: The table is not rendering in preview anymore.

File: source/developer/interactive-dialogs.rst [hunk 6]
```diff
@@ -210,6 +210,36 @@ The list of supported fields for the ``select`` type element is included below:
...
+    :header: "Field", "Type", "Description"
+
+    "display_name", "String", "Display name of the field shown to the user in the dialog. Maximum 24 characters."
+    "name", "String", "Name of the field element used by the integration. Maximum 300 characters. You should use unique “name” fields in the same dialog."
+    "type", "String", "Set this value to ``bool`` for a checkbox element."
+    "optional", "Boolean", "(Optional) Set to true if this form element is not required. Default is ``false``."
+    "help_text", "String", "(Optional) Set help text for this form element. Maximum 150 characters."
+    "default", "String", "(Optional) Set a default value for this form element. True or false".
```
Comment (jasonblais) [inline source/developer/interactive-dialogs.rst]:
```suggestion
    "default", "String", "(Optional) Set a default value for this form element. True or false."
```

Comment (jasonblais) [reply source/developer/interactive-dialogs.rst]: ^this should fix the rendering issue for the table

File: source/developer/interactive-dialogs.rst [hunk 7]
```diff
@@ -210,6 +210,36 @@ The list of supported fields for the ``select`` type element is included below:
     "default", "String", "(Optional) Set a default value for this form element. Maximum 3,000 characters."
     "placeholder", "String", "(Optional) A string displayed to help guide users in completing the element. Maximum 3,000 characters."
 
+Checkbox Element
+^^^^^^^^^^^^^^^^^^^^^^^
+
+In Mattermost version 5.14 and later, you can use checkbox elements. It looks like a plain text field with a checkbox to be selected. Below is an example of a checkbox element that asks for meeting feedback:
```
Comment (jasonblais) [inline source/developer/interactive-dialogs.rst]:
```suggestion
In Mattermost version 5.16 and later, you can use checkbox elements. It looks like a plain text field with a checkbox to be selected. Below is an example of a checkbox element that asks for meeting feedback:
```

Comment (jasonblais) [reply source/developer/interactive-dialogs.rst]: ^this should help resolve Hanzei's note about supported versions

**Comment by jasonblais** [comment]
Hey @hvhallmann, just touching base if you'll have a chance to look at the two outstanding notes above? Otherwise should be ready to merge :)

**Comment by hanzei** [comment]
@aaronrothschild Thoughts on merging is before `v5.16` is released? This way developers can get there hands on the feature and start experimenting with it.

**Comment by aaronrothschild** [comment]
> @aaronrothschild Thoughts on merging is before `v5.16` is released? This way developers can get there hands on the feature and start experimenting with it.

Sure, I don't see a problem with that.  i checked and the new text explicitly mentions v5.16 being needed, so it shouldn't cause confusion IMO.

**Pull request merged**